setupext.py

```diff
@@ -531,6 +540,41 @@ def do_custom_build(self):
         """
         pass
 
+    def install_help_msg(self):
+        """
+        The help message to show if the package is not installed. The help
+        message shown depends on whether some class variables are present.
+        """
+        def _try_managers(*managers):
+            for manager in managers:
+                pkg_name = self.pkg_names.get(manager, None)
+                if pkg_name:
+                    try:
+                        # `shutil.which()` can be used when Python 2.7 support
+                        # is dropped. It is available in Python 3.3+
+                        _ = check_output(["which", manager],
+                                         stderr=subprocess.STDOUT)
+                        return ('Try installing {0} with `{1} install {2}`'
+                                .format(self.name, manager, pkg_name))
+                    except subprocess.CalledProcessError:
+                        pass
+
+        message = None
+        if sys.platform == "win32":
+            url = self.pkg_names.get("windows_url", None)
+            if url:
+                message = ('Please check {0} for instructions to install {1}'
+                           .format(url, self.name))
+        elif sys.platform == "darwin":
+            message = _try_managers("brew", "port")
+        elif sys.platform.startswith("linux"):
+            release = platform.linux_distribution()[0].lower()
```

Comment:
I had a note I wanted to mention here. `platform.linux_distribution()` is deprecated in Py 3.5 and is planned to be removed in Py 3.7. The related bug can be found at https://bugs.python.org/issue1322

I found the package https://pypi.python.org/pypi/distro which makes a pypi package out of this, but I am not sure of a good way to install this to use it in the setup.py. Suggestions ? (`setup_requires` is meant for this, but it's known to be buggy IIRC)